### PR TITLE
Extract controller logic to view provider

### PR DESCRIPTION
**Pull request by ViniTou**
| Question      | Answer
| ------------- | ---
| Tickets       |  [EZP-31651](https://jira.ez.no/browse/EZP-31651) 
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | yes
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-search/blob/master/LICENSE)

## Desc & adminUI PR:

- Moved all code from SearchController into `ViewBuilder` and `ViewProvider`.

- Deprecations and moved `form types`:
https://github.com/ezsystems/ezplatform-admin-ui/pull/1404

#### Checklist:
- [x] Implement tests
- [x] Coding standards (`$ composer fix-cs`)

### CODE REVIEW

File: src/lib/View/SearchListViewFilter.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+     * @throws \Symfony\Component\OptionsResolver\Exception\InvalidOptionsException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     */
+    public function handleSearchForm(FilterViewBuilderParametersEvent $event): void
+    {
+        $controllerAction = $event->getParameters()->get('_controller');
+
+        if (
+            'Ibexa\Platform\Bundle\SearchBundle\Controller\SearchController::searchAction' !== $controllerAction
```
Comment (bdunogier) [inline src/lib/View/SearchListViewFilter.php:68]: If the controller is meant to be overridable, this won't work. If I recall correctly, we usually test on the route and not on the controller.

Comment (ViniTou) [reply src/lib/View/SearchListViewFilter.php:68]: That controller now returns just view from ViewBuilder, I do not see any reason to override it.
Also, I am pretty sure all ViewFilters are checked against controller and not route.

Comment (bdunogier) [reply src/lib/View/SearchListViewFilter.php:68]: Maybe it was the ViewBuilder, not the Filters.

But... if the controller isn't meant to be customized, what is the point of using the View API at all ? :)

Comment (ITernovtsiy) [reply src/lib/View/SearchListViewFilter.php:68]: FYI:
We have internal extended search bundle, which allows admin to export search results.
So, we have controller that returns `StreamedResponse`.
Will be very helpful if controller can be customizable as well.

**Pull request merged**